src/Tribe/Body_Classes.php

```diff
@@ -0,0 +1,367 @@
+				},
+				ARRAY_FILTER_USE_KEY
+			);
+		}
+	}
+
+	/**
+	 * Remove an array of classes from the queue.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string> $classes The classes to remove.
+	 * @return void
+	 */
+	public function remove_classes( array $classes, $queue = 'display' ) {
+		if ( empty( $classes ) || ! is_array( $classes) ) {
+			return;
+		}
+
+		foreach ( $classes as $class ) {
+			$this->remove_class( $class, $queue );
+		}
+	}
+
+	/**
+	 * Adds the enqueued classes to the body class array.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string> $classes An array of body class names.
+	 * @return void
+	 */
+	private function add_body_classes( $classes = [] ) {
+		// Make sure they should be added.
+		if( ! $this->should_add_body_classes( $this->get_class_names(), $classes ) ) {
+			return $classes;
+		}
+
+		return array_merge( $classes, $this->get_class_names() );
+	}
+
+	/**
+	 * Adds the enqueued classes to the body class array.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string> $classes An array of body class names.
+	 * @return void
+	 */
+	private function add_admin_body_classes( $classes = [] ) {
+		// Make sure they should be added.
+		if( ! $this->should_add_body_classes( $this->get_class_names( 'admin' ), $classes, 'admin' ) ) {
+			return $classes;
+		}
+
+		return array_merge( $classes, $this->get_class_names( 'admin' ) );
+	}
+
+	/**
+	 * Should a individual class be added to the queue.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $class The body class we wish to add.
+	 * @return boolean
+	 */
+	private function should_add_body_class_to_queue( string $class, $queue = 'display' ) {
+		global $post;
+		// default to false!
+		$add = false;
+		// If we are doing an event query, or on an event single, set to true.
+		if (
+			tribe_is_event_query()
+			|| ( $post instanceof \WP_Post && has_shortcode( $post->post_content, 'tribe_events' ) )
+		) {
+			$add = true;
+		}
+
+		if (
+			'admin' === $queue
+			&& ! is_admin()
+		) {
+			$add = false;
+		}
+
+		/**
+		 * Filter whether to add the body class to the queue or not.
+		 *
+		 * @since TBD
+		 *
+		 * @param boolean Whether to add the class to the queue or not.
+		 * @param array $class The array of body class names to add.
+		 */
+		return apply_filters( 'tribe_should_add_body_class_to_queue', $add, $class, $queue );
+	}
+
+	/**
+	 * Logic for whether the body classes, as a whole, should be added.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $add_classes      An array of body class names to add.
+	 * @param array $existing_classes An array of existing body class names from WP.
+	 * @return boolean
+	 */
+	private function should_add_body_classes( array $add_classes, array $existing_classes, $queue = 'display' ) {
+		global $post;
+		// default to false!
+		$add = false;
+		// If we are doing an event query, or on an event single, set to true.
+		if (
+			tribe_is_event_query()
+			|| ( $post instanceof \WP_Post && has_shortcode( $post->post_content, 'tribe_events' ) )
```

Comment:
Move to Pro.

Reply:
https://github.com/moderntribe/events-pro/pull/1565